dot/parachain/types/types.go

```diff
@@ -64,10 +64,30 @@ func (info GroupRotationInfo) GroupForCore(coreIndex CoreIndex, numOfCores uint)
 	return GroupIndex(idx)
 }
 
-func (gri GroupRotationInfo) CoreForGroup(groupIndex GroupIndex, cores uint8) CoreIndex {
-	//nolint
-	// TODO: https://github.com/paritytech/polkadot-sdk/blob/aa68ea58f389c2aa4eefab4bf7bc7b787dd56580/polkadot/primitives/src/v6/mod.rs#L877
-	return CoreIndex{}
+// CoreForGroup returns the index of the group assigned to the given core. This does no checking or
+// whether the group index is in-bounds.
+//
+// `coreIndex` should be less than `numOfCores`, which is capped at `MaxUint32`.
+func (info GroupRotationInfo) CoreForGroup(groupIndex GroupIndex, numOfCores uint) CoreIndex {
+	if info.GroupRotationFrequency == 0 {
+		return CoreIndex{Index: uint32(groupIndex)}
+	}
+	if numOfCores == 0 {
+		return CoreIndex{Index: 0}
+	}
+
+	numOfCores = min(numOfCores, math.MaxUint32)
+
+	var blocksSinceStart uint32
+	if info.Now > info.SessionStartBlock {
+		blocksSinceStart = uint32(info.Now - info.SessionStartBlock)
+	}
+
+	rotations := blocksSinceStart / uint32(info.GroupRotationFrequency)
```

Comment:
why are we calculating using `uint32`?

Reply:
```
// BlockNumber The block number type.
type BlockNumber uint32

// GroupRotationInfo A helper data-type for tracking validator-group rotations.
type GroupRotationInfo struct {
	// SessionStartBlock is the block number at which the session started
	SessionStartBlock BlockNumber `scale:"1"`
	// GroupRotationFrequency indicates how often groups rotate. 0 means never.
	GroupRotationFrequency BlockNumber `scale:"2"`
	// Now indicates the current block number.
	Now BlockNumber `scale:"3"`
}
```

All fields are of uint32 type. In addition we are calculating the core index which is also uint32